src/circuit/gadgets/constraints.rs

```diff
@@ -785,15 +777,23 @@ pub fn enforce_less_than_bound<F: LurkField, CS: ConstraintSystem<F>>(
 pub fn enforce_is_negative<F: LurkField, CS: ConstraintSystem<F>>(
     mut cs: CS,
     num: &AllocatedNum<F>,
+    is_strict: bool,
 ) -> Result<Boolean, SynthesisError> {
     let double_num = add(
         &mut cs.namespace(|| "double num"),
         &num,
         &num,
     )?;
-    let double_num_bits = double_num
-        .to_bits_le_strict(&mut cs.namespace(|| "double num bits"))
-        .unwrap();
+    let double_num_bits: Vec<Boolean> = if is_strict {
+        double_num
+            .to_bits_le_strict(&mut cs.namespace(|| "double num bits"))
+            .unwrap()
+    } else {
+        double_num
+        .to_bits_le(&mut cs.namespace(|| "double num bits"))
+        .unwrap()
+    };
```

Comment:
Does it actually make sense to support this? If `is_strict` is false and `to_bits_le()` (not strict) is used? Then does `enforce_is_negative` actually do what the name implies?

Reply:
This function is confusingly named. It does *not* enforce that `num` is negative. Rather, it allocates a `Boolean` which is true iff `num` is negative.

Reply:
renamed